### PR TITLE
Feature/highlight a11y violation on component

### PR DESCRIPTION
**Pull request by CodeByAlex**
Issue: #6001

## What I did

I added highlight toggles for each accessibility rule and category at every level. A user can now click on the highlight checkbox and see their violation, pass, or incomplete rule highlighted in the iframe. Each type (violation, pass, or incomplete) has a corresponding color to differentiate the rule when highlighting.

Design:

Highlight-Toggle component - can take in one to many elements that it can highlight. This means that the toggle component can be placed at any level and highlight the element or elements passed in based on a switch.

Redux is being used to keep track of a global map where the key is the element itself and the value is an object containing the toggle state of the element and the original element outline so that the component isn't changed when the highlight is removed.

The highlight toggle uses the redux map to keep track of whether or not the element or elements within it are highlighted and whether or not the toggle state is true.

I worked with @domyen on the design elements for this feature to keep consistency across the app.

## How to test

- Is this testable with Jest or Chromatic screenshots? yes
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? Maybe to describe the feature?

a11y-highlight-feature

### CODE REVIEW

**Comment by domyen** [comment]
Great work @CodeByAlex! QAing this. 

After playing with this in real life, I'd suggest these layout changes:
Screen Shot 2019-03-24 at 10 29 37 AM
I think if we align the checkboxes to the right and underneath the "highlight results" it becomes obvious what they're about (similar to column headers in a table). That means we can get rid of the repeating "highlight" label on each line item.

I can take a stab at it if you like. What do you think?

**Comment by CodeByAlex** [comment]
Thanks @domyen! Thanks for the feedback. I'll make sure that the "Highlight results" and checkbox are vertically centered and uncapitalize "results". My only concern about aligning the checkboxes to the right would be in scenarios where people are using wide screens. It may be hard to determine which checkbox is associated with which rule (one above or below in most cases). What do you think?

**Comment by domyen** [comment]
👍 We can solve this by showing the boundaries of the row:  
image

The right hand alignment makes the highlight checkbox a bit less distracting. After all the goal is to communicate the rules not that there's a checkbox. :)

And while we're here:
- The button should extend to the full length of the row
- The button should have a `cursor:pointer`
- button:hover should have bg color `rgba(0,0,0,.05)`

You're a legend @CodeByAlex 🙏 🙇

**Comment by CodeByAlex** [comment]
I like that a lot @domyen , Ill make that design change

**Comment by CodeByAlex** [comment]
@domyen, Ive made the changes. Let me know what you think
a11y-highlight-feature-mod

**Comment by CodeByAlex** [comment]
I think one other touch-up that would make this look a little sleeker is putting the "Highlight results" text on the left side of the checkbox. This keeps a uniform line down the right side. Would you agree @domyen?

image

**Review by domyen — APPROVED** [review]
LGTM from a UI perspective

**Review by shilman — APPROVED** [review]
This looks amazing! Great work @CodeByAlex and @domyen! Esp love the right aligned checkbox on top -- details ❤️❤️❤️ 👌 !

Since the UI has gotten complex, and it's been awhile since I used redux for state management (cc @alexandrebodin @ndelangen), here's what I'd like to see on the code side:
- [ ] Stories for the major components
- [ ] Hooked into `official-storybook` or another top-level storybook to simplify testing

This might be outside the scope of this PR, but we should probably do this for all the core addons. It bothers me to see Jest tests with snapshots when we've already built a much nicer way to do this inside Storybook.

**Comment by ndelangen** [comment]
@CodeByAlex can you please allow maintainers to push to your branch? I'd like to fix the merge conflicts and merge

**Comment by CodeByAlex** [comment]
Sure thing @ndelangen

**Comment by ndelangen** [comment]
Thank you! Let's get this merged, amazing feature!

**Comment by CodeByAlex** [comment]
Thanks, @ndelangen!

File: addons/notes/src/Panel.tsx
Comment (Armanio) [inline addons/notes/src/Panel.tsx]: Why it is here? 🤔

File: lib/api/src/version.ts
Comment (Armanio) [inline lib/api/src/version.ts]: And what is it? Does it need in this PR?

File: addons/notes/src/Panel.tsx
Comment (CodeByAlex) [reply addons/notes/src/Panel.tsx]: Not Exactly sure why and I’d have to look at it to figure it out but it looks like a format change

**Comment by Armanio** [comment]
Looking great! 
I wrote a few comments. Check it before you merge it, please. 
Thank you for your effort.

File: lib/api/src/version.ts
Comment (CodeByAlex) [reply lib/api/src/version.ts]: @ndelangen bumped the version here during the merge

File: addons/notes/src/Panel.tsx
Comment (ndelangen) [reply addons/notes/src/Panel.tsx]: Just a formatting change, I guess it was a warning as didn't cause the build to fail before, but got fixed by accident/automatically

**Pull request merged**
